File: jwt-login/backend-springboot/src/main/java/backend_springboot/config/exception/GlobalExceptionHandler.java
package backend_springboot.config.exception;

import backend_springboot.config.utils.ErrorMessageUtil;
import backend_springboot.domain.auth.exception.AuthException;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
@RequiredArgsConstructor
public class GlobalExceptionHandler {

    private final ErrorMessageUtil errorMessageUtil;

    @ExceptionHandler(DomainException.class)
    public ResponseEntity<String> handleAuthException(DomainException e) {
        HttpStatus httpStatus = e.getHttpStatus() != null ? e.getHttpStatus() : HttpStatus.UNAUTHORIZED;
        String errorMessage = errorMessageUtil.getMessage(e.getMessage());
        System.out.println("errorMessage = " + errorMessage);
        return ResponseEntity
                .status(httpStatus)
                .body(errorMessage);
    }
}

File: couponwithlock/src/main/kotlin/blog/couponwithlock/core/exception/GlobalExceptionHandler.kt
package blog.couponwithlock.core.exception

import blog.couponwithlock.core.exception.error.DomainException
import blog.couponwithlock.core.presentation.ErrorResponse
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(DomainException::class)
    fun handleDomainException(ex: DomainException): ResponseEntity<ErrorResponse> {
        val httpStatus = ex.httpStatus
        val errorResponse = ErrorResponse.createDomainErrorResponse(httpStatus.value(), ex)
        return ResponseEntity.status(httpStatus).body(errorResponse)
    }
}
